### PR TITLE
Add `2.3.0-1-dev`

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Create `AC 2.3.0-1-dev`

### CODE REVIEW

**Review by ashb — APPROVED** [review]

File: 2.3.0/bullseye/include/entrypoint [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+url_parse_regex="[^:]+://([^@/]*@)?([^/:]*):?([0-9]*)/?"
+
+# Wait for postgres then init the db
+if [[ -n $AIRFLOW__CORE__SQL_ALCHEMY_CONN  ]]; then
+  # Wait for database port to open up
+  [[ ${AIRFLOW__CORE__SQL_ALCHEMY_CONN} =~ $url_parse_regex ]]
+  HOST=${BASH_REMATCH[2]}
+  PORT=${BASH_REMATCH[3]}
+  echo "Waiting for host: ${HOST} ${PORT}"
```
Comment (blag) [inline 2.3.0/bullseye/include/entrypoint:30]:
```suggestion
  # Wait for database port to open up
  [[ ${AIRFLOW__CORE__SQL_ALCHEMY_CONN} =~ $url_parse_regex ]]
  HOST=${BASH_REMATCH[2]}
  PORT=${BASH_REMATCH[3]}
  echo "Waiting for database: ${HOST}:${PORT}"
```

File: 2.3.0/bullseye/include/entrypoint [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    sleep 0.001
+  done
+fi
+
+if [[ -n $AIRFLOW__CELERY__BROKER_URL ]] && [[ $CMD =~ ^(scheduler|celery worker|celery flower)$ ]]; then
+  # Wait for database port to open up
+  [[ ${AIRFLOW__CELERY__BROKER_URL} =~ $url_parse_regex ]]
+  HOST=${BASH_REMATCH[2]}
+  PORT=${BASH_REMATCH[3]}
+  echo "Waiting for host: ${HOST} ${PORT}"
```
Comment (blag) [inline 2.3.0/bullseye/include/entrypoint:41]:
```suggestion
  # Wait for broker port to open up
  [[ ${AIRFLOW__CELERY__BROKER_URL} =~ $url_parse_regex ]]
  HOST=${BASH_REMATCH[2]}
  PORT=${BASH_REMATCH[3]}
  echo "Waiting for broker: ${HOST}:${PORT}"
```

File: 2.3.0/bullseye/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    # Run pods spun up by Kubernetes Executor as astro user
+    -e 's/^run_as_user =.*/run_as_user = 50000/g' \
+    # Needed by astronomer-version-check-plugin
+    -e 's/^lazy_load_plugins =.*/lazy_load_plugins = False/g' \
+    # We sync permissions in the entrypoint so do not need to run in the Webserver again
+    -e 's/^update_fab_perms =.*/update_fab_perms = False/g' \
+    # Use Auth Backend defined in Astronomer FAB Security Manager
+    -e 's/^auth_backend =.*/auth_backend = astronomer.flask_appbuilder.current_user_backend/g' \
+    # The number of seconds to wait before timing out send_task_to_executor or fetch_celery_task_state operations.
+    -e 's/^operation_timeout =.*/operation_timeout = 10.0/g' \
```
Comment (blag) [inline 2.3.0/bullseye/Dockerfile:177]:
```suggestion
# Run pods spun up by Kubernetes Executor as astro user
# Lazily load all plugins, for astronomer-version-check-plugin
# Sync permissions in the entrypoint so we do not need to run in the Webserver again
# Use Astronomer FAB Security Manager authentication backend
# Configure a 10.0s timeout for send_task_to_executor or fetch_celery_task_state operations.
RUN sed -i \
    -e 's/^run_as_user =.*/run_as_user = 50000/g' \
    -e 's/^lazy_load_plugins =.*/lazy_load_plugins = False/g' \
    -e 's/^update_fab_perms =.*/update_fab_perms = False/g' \
    -e 's/^auth_backend =.*/auth_backend = astronomer.flask_appbuilder.current_user_backend/g' \
    -e 's/^operation_timeout =.*/operation_timeout = 10.0/g' \
```

See #463.

**Review by blag — CHANGES_REQUESTED** [review]
Just a few nitpicks, otherwise looks good!

I differentiated the error messages for waiting for the database vs. waiting for the container, that should hopefully make it easier to troubleshoot if one of those isn't reachable.

**Comment by ephraimbuddy** [comment]
We need to create epoch providers. We don't have these ones in the constraints file

File: 2.3.0/bullseye/build-time-pip-constraints.txt [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+apache-airflow-providers-apache-kylin==1!2.0.4
+apache-airflow-providers-apache-livy==1!2.2.3
+apache-airflow-providers-apache-pig==1!2.0.4
+apache-airflow-providers-apache-pinot==1!2.0.4
+apache-airflow-providers-apache-spark==1!2.1.3
+apache-airflow-providers-apache-sqoop==1!2.1.3
+apache-airflow-providers-asana==1!1.1.3
+apache-airflow-providers-celery==1!2.1.4
+apache-airflow-providers-cloudant==1!2.0.4
+apache-airflow-providers-cncf-kubernetes==1!3.1.2
```
Comment (jedcunningham) [inline 2.3.0/bullseye/build-time-pip-constraints.txt:68]:
```suggestion
apache-airflow-providers-cncf-kubernetes==1!4.0.0
```

**Review by jedcunningham — APPROVED** [review]

File: 2.3.0/bullseye/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        libsasl2-dev \
+        libssl-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+ARG VERSION="2.3.0-1-*"
+ARG SUBMODULES="async,azure,amazon,elasticsearch,google,password,cncf.kubernetes,mysql,postgres,redis,slack,ssh,statsd,virtualenv"
+ARG AIRFLOW_MODULE="astronomer_certified[${SUBMODULES}]==$VERSION"
+ARG AIRFLOW_VERSION="2.3.0"
+ARG ASTRONOMER_FAB_SECURITY_MANAGER_VERSION="1.8.4"
```
Comment (jedcunningham) [inline 2.3.0/bullseye/Dockerfile:117]:
```suggestion
ARG ASTRONOMER_FAB_SECURITY_MANAGER_VERSION="1.9.0"
```

File: 2.3.0/bullseye/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+ARG PACKAGE_UPGRADE_EPOCH_NUMBER="1"
+
+RUN apt-get update \
+    && apt-get upgrade -y --no-install-recommends \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+ARG VERSION="2.3.0-1-*"
+ARG AIRFLOW_VERSION="2.3.0"
+ARG ASTRONOMER_FAB_SECURITY_MANAGER_VERSION="1.8.4"
```
Comment (jedcunningham) [inline 2.3.0/bullseye/Dockerfile:151]:
```suggestion
ARG ASTRONOMER_FAB_SECURITY_MANAGER_VERSION="1.9.0"
```

File: 2.3.0/bullseye/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+# Run pods spun up by Kubernetes Executor as astro user
+# Lazily load all plugins, for astronomer-version-check-plugin
+# Sync permissions in the entrypoint so we do not need to run in the Webserver again
+# Use Astronomer FAB Security Manager authentication backend
+# Configure a 10.0s timeout for send_task_to_executor or fetch_celery_task_state operations.
+RUN sed -i \
+    -e 's/^run_as_user =.*/run_as_user = 50000/g' \
+    -e 's/^lazy_load_plugins =.*/lazy_load_plugins = False/g' \
+    -e 's/^update_fab_perms =.*/update_fab_perms = False/g' \
+    -e 's/^auth_backend =.*/auth_backend = astronomer.flask_appbuilder.current_user_backend/g' \
```
Comment (jedcunningham) [inline 2.3.0/bullseye/Dockerfile:176]:
```suggestion
    -e 's/^auth_backends =.*/auth_backends = astronomer.flask_appbuilder.current_user_backend/g' \
```

**Pull request merged**